File: TotalCrossSDK/src/totalcross/phone/SMS.java
/*********************************************************************************
 *  TotalCross Software Development Kit                                          *
 *  Copyright (C) 2000-2012 SuperWaba Ltda.                                      *
 *  All Rights Reserved                                                          *
 *                                                                               *
 *  This library and virtual machine is distributed in the hope that it will     *
 *  be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of    *
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.                         *
 *                                                                               *
 *  This file is covered by the GNU LESSER GENERAL PUBLIC LICENSE VERSION 3.0    *
 *  A copy of this license is located in file license.txt at the root of this    *
 *  SDK or can be downloaded here:                                               *
 *  http://www.gnu.org/licenses/lgpl-3.0.txt                                     *
 *                                                                               *
 *********************************************************************************/



package totalcross.phone;

/** Used to send and receive SMS messages. Currently supports only WP8.
 * See the SmsSample for an example.
 */
public class SMS
{
   /** Sends the given message to the destination.
    * @param destination The number to send to.
    * @param message The message to send. Note that some phones will strip unicode characters.
    * @throws totalcross.io.IOException if an error occurs.
    */
   public static void send(String destination, String message) throws totalcross.io.IOException
   {
   }
   
   /** Supposed to block until a message becomes available.
    *  Does not work on the currently supported platforms.
    *  @deprecated
    */
   public static String[] receive() throws totalcross.io.IOException
   {
      return null;
   }
}

File: TotalCrossVM/src/nm/phone/SMS.c
// Copyright (C) 2000-2013 SuperWaba Ltda.
// Copyright (C) 2014-2020 TotalCross Global Mobile Platform Ltda. 
//
// TotalCross Software Development Kit
//
// This library and virtual machine is distributed in the hope that it will
// be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
//
// SPDX-License-Identifier: LGPL-2.1-only

#include "tcvm.h"

#if defined WP8

#elif defined (WIN32) || defined (WINCE)
 #include "win/SMS_c.h"
#endif

//////////////////////////////////////////////////////////////////////////
TC_API void tpSMS_send_ss(NMParams p) // totalcross/phone/SMS native public static void send(String destination, String message) throws totalcross.io.IOException;
{
#if defined (WINCE) || defined (WP8)
   TCObject destination = p->obj[0];
   TCObject message = p->obj[1];
   if (destination == null)
      throwNullArgumentException(p->currentContext, "destination");
   else
   if (message == null)
      throwNullArgumentException(p->currentContext, "message");
   else
   {
#ifdef WINCE
      TCHARP szMessage, szDestination;

      szMessage = String2TCHARP(message);
      szDestination = String2TCHARP(destination);
      if (!szMessage || !szDestination)
         throwException(p->currentContext, OutOfMemoryError, !szMessage?"When allocating 'message'":"'When allocating 'destination'");
      else
         SmsSend(p->currentContext, szMessage, szDestination);
#elif defined (WP8)
      JCharP szMessage = JCharPDup(String_charsStart(message), String_charsLen(message));
      JCharP szDestination = JCharPDup(String_charsStart(destination), String_charsLen(destination));

      if (!szMessage || !szDestination)
         throwException(p->currentContext, OutOfMemoryError, !szMessage ? "When allocating 'message'" : "'When allocating 'destination'");
      else
         smsSendCPP(szMessage, szDestination);
         
#endif
      xfree(szMessage);
      xfree(szDestination);
   }
#else
   p = 0;
#endif
}
//////////////////////////////////////////////////////////////////////////
TC_API void tpSMS_receive(NMParams p) // totalcross/phone/SMS native public static String[] receive() throws totalcross.io.IOException;
{
#if defined (WINCE)
   SmsReceive(p->currentContext, &p->retO);
#else
   p->retO = NULL;
#endif
}

#ifdef ENABLE_TEST_SUITE
//#include "SMS_test.h"
#endif
